tpl/template_funcs.go

```diff
@@ -239,6 +246,127 @@ func slicestr(a interface{}, startEnd ...interface{}) (string, error) {
 
 }
 
+func truncate(a interface{}, options ...interface{}) (template.HTML, error) {
+	length, err := cast.ToIntE(a)
+	if err != nil {
+		return "", err
+	}
+	var textParam interface{}
+	var text string
+	var ellipsis template.HTML
+
+	switch len(options) {
+	case 0:
+		return "", errors.New("truncate requires a length and a string")
+	case 1:
+		textParam = options[0]
+		ellipsis = " …"
+	case 2:
+		textParam = options[1]
+		var ok bool
+		if ellipsis, ok = options[0].(template.HTML); !ok {
+			s, e := cast.ToStringE(options[0])
+			if e != nil {
+				return "", errors.New("ellipsis must be a string")
+			}
+			ellipsis = template.HTML(html.EscapeString(s))
+		}
+	default:
+		return "", errors.New("too many arguments passed to truncate")
+	}
+	if err != nil {
+		return "", errors.New("text to truncate must be a string")
+	}
+	text, err = cast.ToStringE(textParam)
+	if err != nil {
+		return "", errors.New("text must be a string")
+	}
+
+	if html, ok := textParam.(template.HTML); ok {
+		return truncateHTML(length, ellipsis, html)
+	}
+
+	if len(text) <= length {
+		return template.HTML(html.EscapeString(text)), nil
+	}
+
+	var count int
+	words := strings.Fields(text)
```

Comment:
I'd bet that something like [this](https://github.com/spf13/hugo/blob/19b6fdf2b1a03ca49d754810dd4d9d414e59ef1a/helpers/content.go#L469) would be more efficient.  Doesn't use `strings.Fields` that would generate more allocs.